### PR TITLE
oded/tfl-962

### PR DESCRIPTION
**Pull request by odeds**

### CODE REVIEW

File: src/components/diagnosticExercise/config/route.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                        currentState: function currentState(WorkoutsDiagnosticFlow, $stateParams) {
+                            'ngInject';
+                            return WorkoutsDiagnosticFlow.getDiagnosticFlowCurrentState(null, $stateParams.skipIntro);
+                        }
+                    },
+                    controller: 'WorkoutsDiagnosticController',
+                    controllerAs: 'vm'
+                })
+                .state('diagnostic.intro', {
+                    url: '/intro/:id/:subjectId/:order',
```
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: what id, order stands for ?

File: src/components/diagnosticExercise/config/route.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                    templateUrl: 'components/diagnosticExercise/templates/workoutsDiagnosticIntro.template.html',
+                    controller: 'WorkoutsDiagnosticIntroController',
+                    controllerAs: 'vm'
+                })
+                .state('diagnostic.exercise', {
+                    url: '/exercise/:id/:sectionId',
```
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: Do we really need this url ?

File: src/components/diagnosticExercise/config/route.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                                } else {
+                                    questionResults.pop();
+                                    delete questionResults[questionResults.length - 1].userAnswer;
+                                }
+                                exerciseData.resultsData.$save();
+                            }
+                    }
+                })
+                .state('diagnostic.preSummary', {
+                    url: '/preSummary',
```
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: Do we really need this url ?

File: src/components/diagnosticExercise/config/route.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                    controller: ['$timeout', '$state', function ($timeout, $state) {
+                        var VIDEO_DURATION = 6000;
+                        $timeout(function () {
+                            $state.go('diagnostic.summary');
+                        }, VIDEO_DURATION);
+                    }],
+                    controllerAs: 'vm'
+                })
+                .state('diagnostic.summary', {
+                    url: '/summary',
```
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: Do we really need this url ?

File: src/components/diagnosticExercise/svg/flag-icon.svg [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<svg x="0px"
+     y="0px"
+	 viewBox="-145 277 60 60"
+	 class="flag-svg">
+    <style type="text/css">
+        .flag-svg .st0 {
```
Comment (igormas) [inline src/components/diagnosticExercise/svg/flag-icon.svg]: add init width to this svg

File: src/components/userGoals/services/userGoals.service.js [hunk 6]
```diff
@@ -2,10 +2,11 @@
 
 angular.module('znk.infra-web-app.userGoals').provider('UserGoalsService', [function() {
 
-        this.$get = ['InfraConfigSrv', 'StorageSrv', '$q', function (InfraConfigSrv, StorageSrv, $q) {
+        this.$get = ['InfraConfigSrv', 'StorageSrv', '$q', 'ScoringService', function (InfraConfigSrv, StorageSrv, $q, ScoringService) {
             var self = this;
             var goalsPath = StorageSrv.variables.appUserSpacePath + '/goals';
-            var defaultSubjectScore = self.settings.defaultSubjectScore;
+            var scoringSettings = ScoringService.getScoringSettings();
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: Please change getScoringSettings -> getExamScoreLimits

File: src/components/userGoals/services/userGoals.service.js [hunk 7]
```diff
@@ -2,10 +2,11 @@
...
+            var defaultSubjectScore = scoringSettings.defaultSubjectScore;
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: What defaultSubjectScore stands for ?

File: src/components/userGoalsSelection/directives/userGoals.directive.js [hunk 8]
```diff
@@ -1,19 +1,21 @@
...
                     onSave: '&?',
                     setting: '='
                 },
                 link: function link(scope) {
-                    $translatePartialLoader.addPart('userGoals');
+                    $translatePartialLoader.addPart('userGoalsSelection');
                     var userGoalRef;
-                    scope.goalsSettingsFromSrv = UserGoalsService.getGoalsSettings();
+                    var scoringSettings = UserGoalsService.getScoringSettings();
+                    var goalsSettings = UserGoalsService.getGoalsSettings();
```
Comment (igormas) [inline src/components/userGoalsSelection/directives/userGoals.directive.js]: its too general name....

File: src/components/userGoalsSelection/directives/userGoals.directive.js [hunk 9]
```diff
@@ -1,19 +1,21 @@
...
                 scope: {
                     onSave: '&?',
                     setting: '='
                 },
                 link: function link(scope) {
-                    $translatePartialLoader.addPart('userGoals');
+                    $translatePartialLoader.addPart('userGoalsSelection');
                     var userGoalRef;
-                    scope.goalsSettingsFromSrv = UserGoalsService.getGoalsSettings();
+                    var scoringSettings = UserGoalsService.getScoringSettings();
```
Comment (igormas) [inline src/components/userGoalsSelection/directives/userGoals.directive.js]: Lets discuss

File: src/components/userGoalsSelection/main.scss [hunk 10]
```diff
@@ -3,10 +3,10 @@
         './bower_components/angular-material/angular-material.scss',
         './bower_components/infra/dist/mixins/mixins',
         './src/scss/imports/angular-material-override',
-        './scss/ng-tag-input-override',
-        './scss/user-goals-directive',
-        './scss/goal-select-directive',
-        './scss/school-select-directive';
+'scss/ng-tag-input-override',
```
Comment (igormas) [inline src/components/userGoalsSelection/main.scss]: align back

File: src/components/userGoalsSelection/svg/dropdown-arrow.svg [hunk 11]
```diff
@@ -0,0 +1,6 @@
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" viewBox="0 0 242.8 117.4" class="dropdown-arrow-icon-svg">
+<style type="text/css">
+	.dropdown-arrow-icon-svg .st0{fill:none;stroke:#000000;stroke-width:18;stroke-linecap:round;stroke-linejoin:round;stroke-miterlimit:10;}
```
Comment (igormas) [inline src/components/userGoalsSelection/svg/dropdown-arrow.svg]: align
add init width or height

File: src/components/userGoalsSelection/templates/userGoals.template.html [hunk 12]
```diff
@@ -35,8 +35,8 @@
                     </div>
                     <span class="subject-title" translate=".{{subject.name | uppercase}}"></span>
                     <goal-select
-                        min-score="goalsSettingsFromSrv.minGoalsScore"
-                        max-score="goalsSettingsFromSrv.maxGoalsScore"
+                        min-score="goalsSettingsFromSrv.subjectMinScore"
```
Comment (igormas) [inline src/components/userGoalsSelection/templates/userGoals.template.html]: Wont it be better min-score="sectionScoreLimit.min" ?

File: src/components/diagnosticExercise/config/route.js (hunk 1 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: id of section, order in flow 
ye

File: src/components/diagnosticExercise/config/route.js (hunk 2 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: i need the id of exam and sectionId also

File: src/components/diagnosticExercise/config/route.js (hunk 3 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: no, removed it

File: src/components/diagnosticExercise/svg/flag-icon.svg (hunk 5 above)
Comment (odeds) [inline src/components/diagnosticExercise/svg/flag-icon.svg]: done

File: src/components/userGoals/services/userGoals.service.js (hunk 7 above)
Comment (odeds) [inline src/components/userGoals/services/userGoals.service.js]: defaultSubjectScore goal like 600 for sat

File: src/components/userGoalsSelection/main.scss (hunk 10 above)
Comment (odeds) [inline src/components/userGoalsSelection/main.scss]: done

File: src/components/userGoalsSelection/svg/dropdown-arrow.svg (hunk 11 above)
Comment (odeds) [inline src/components/userGoalsSelection/svg/dropdown-arrow.svg]: done

File: src/components/userGoalsSelection/templates/userGoals.template.html (hunk 12 above)
Comment (odeds) [inline src/components/userGoalsSelection/templates/userGoals.template.html]: changed the goalsSettingsFromSrv to sectionScoreSettings

File: src/components/diagnosticExercise/config/route.js (hunk 1 above)
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: so change to sectionId why do we need the order in the url ?
Why we need the url here ?

File: src/components/diagnosticExercise/config/route.js (hunk 2 above)
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: It should come from the parent state.

File: src/components/diagnosticExercise/config/route.js [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
```
Comment (igormas) [inline src/components/diagnosticExercise/config/route.js]: no need for url...

File: src/components/diagnosticExercise/controllers/workoutsDiagnostic.controller.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra-web-app.diagnosticExercise').controller('WorkoutsDiagnosticController', function($state, currentState, $translatePartialLoader) {
+        'ngInject';
+
+        var EXAM_STATE = 'diagnostic';
```
Comment (igormas) [inline src/components/diagnosticExercise/controllers/workoutsDiagnostic.controller.js]: Exam state is diagnostic ?

File: src/components/diagnosticExercise/svg/flag-icon.svg (hunk 5 above)
Comment (igormas) [inline src/components/diagnosticExercise/svg/flag-icon.svg]: Are you the width should be applied to .st0 class and not .flag-svg ?

File: src/components/userGoals/services/userGoals.service.js [hunk 15]
```diff
@@ -41,8 +42,8 @@ angular.module('znk.infra-web-app.userGoals').provider('UserGoalsService', [func
                 // 2. Calc the average score for each school and set it for each subject goal
 
                 return userGoalsServiceObj.getGoals().then(function (userGoals) {
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: What userGoalsServiceObj.calcCompositeScore do ?

File: src/components/userGoals/services/userGoals.service.js [hunk 16]
```diff
@@ -41,8 +42,8 @@ angular.module('znk.infra-web-app.userGoals').provider('UserGoalsService', [func
                 // 2. Calc the average score for each school and set it for each subject goal
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: This calculation belongs to ACT

File: src/components/userGoals/services/userGoals.service.js [hunk 17]
```diff
@@ -41,8 +42,8 @@ angular.module('znk.infra-web-app.userGoals').provider('UserGoalsService', [func
...
 
                 return userGoalsServiceObj.getGoals().then(function (userGoals) {
-                    var minSchoolScore = self.settings.minSchoolScore,
-                        maxSchoolScore = self.settings.maxSchoolScore,
+                    var minSchoolScore = scoringLimits.exam.min,
+                        maxSchoolScore = scoringLimits.exam.max,
                         avgScores = [];
 
                     angular.forEach(userSchools, function (school) {
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: calculation which belongs to ACT

File: src/components/userGoals/services/userGoals.service.js [hunk 18]
```diff
@@ -78,10 +79,14 @@ angular.module('znk.infra-web-app.userGoals').provider('UserGoalsService', [func
                  return self.settings;
             };
 
+            userGoalsServiceObj.getScoringLimits = function() {
```
Comment (igormas) [inline src/components/userGoals/services/userGoals.service.js]: It should be exposed by scoring service

File: src/components/userGoalsSelection/services/userSchools.service.js [hunk 19]
```diff
@@ -1,6 +1,6 @@
 'use strict';
 
-angular.module('znk.infra-web-app.userGoals').service('UserSchoolsService', ['InfraConfigSrv', 'StorageSrv', 'ENV', '$http', 'UserGoalsService', '$q',
+angular.module('znk.infra-web-app.userGoalsSelection').service('UserSchoolsService', ['InfraConfigSrv', 'StorageSrv', 'ENV', '$http', 'UserGoalsService', '$q',
```
Comment (igormas) [inline src/components/userGoalsSelection/services/userSchools.service.js]: This service should be named userGoalsSelectionService

File: src/scss/imports/_colors.scss [hunk 20]
```diff
@@ -40,6 +43,11 @@ $bgColor26: rgba(10, 155, 173, 0.6);
 $bgColor27: #f9f9f9;
 $bgColor28: #f2f2f2;
 $bgColor29: #f8f8f8;
+$bgColor30: #CACACA;
+$bgColor31: #acacac;
+$bgColor32: #f9f9f9;
+$bgColor33: #c9c9c9;
+$bgColor24: #81649c;
```
Comment (igormas) [inline src/scss/imports/_colors.scss]: bgColor24 ?

File: src/components/diagnosticExercise/controllers/workoutsDiagnostic.controller.js (hunk 14 above)
Comment (odeds) [inline src/components/diagnosticExercise/controllers/workoutsDiagnostic.controller.js]: ye that's the main route

File: src/components/diagnosticExercise/config/route.js (hunk 1 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: remove url

File: src/components/diagnosticExercise/config/route.js (hunk 2 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: remove url

File: src/components/diagnosticExercise/config/route.js (hunk 13 above)
Comment (odeds) [inline src/components/diagnosticExercise/config/route.js]: removed url

File: src/components/userGoals/services/userGoals.service.js (hunk 15 above)
Comment (odeds) [inline src/components/userGoals/services/userGoals.service.js]: now required from config phase

File: src/components/userGoals/services/userGoals.service.js (hunk 16 above)
Comment (odeds) [inline src/components/userGoals/services/userGoals.service.js]: config phase

File: src/components/userGoals/services/userGoals.service.js (hunk 17 above)
Comment (odeds) [inline src/components/userGoals/services/userGoals.service.js]: config phase

File: src/components/userGoals/services/userGoals.service.js (hunk 18 above)
Comment (odeds) [inline src/components/userGoals/services/userGoals.service.js]: changed

File: src/components/userGoalsSelection/services/userSchools.service.js (hunk 19 above)
Comment (odeds) [inline src/components/userGoalsSelection/services/userSchools.service.js]: changed

File: src/scss/imports/_colors.scss (hunk 20 above)
Comment (odeds) [inline src/scss/imports/_colors.scss]: fixed to $bgColor34

**Pull request merged**